versioned/tiered/dynamodb/src/main/java/com/dremio/nessie/versioned/store/dynamo/DynamoL1.java

```diff
@@ -118,38 +117,35 @@ private void addKeysSafe(String key, AttributeValue value) {
    * Deserialize a DynamoDB entity into the given consumer.
    */
   static void toConsumer(Map<String, AttributeValue> entity, L1 consumer) {
-    consumer.id(deserializeId(entity, ID)).dt(AttributeValueUtil.getDt(entity));
+    consumer = consumer.id(deserializeId(entity, ID)).dt(AttributeValueUtil.getDt(entity));
```

Comment:
rather than force this pattern, please update the javadoc on BaseValue to clarify that the consumer should hold shared state e.g. the user can call either the returned value or the original value and the same thing happens.